### ISSUE TITLE
[Android] NPE in Filament DisplayHelper.updateDisplayInfo when closing a SceneView screen

### ISSUE THREAD

**Issue by michihiro-ikeda-safie** [sceneview/sceneview#2709]
### Platform

Android

### Module

sceneview (3D only)

### SceneView version

4.17.0

### Device / OS

Various phones — predominantly Android 16 devices with adaptive refresh displays

### What happened?

A `NullPointerException` crash inside Filament's `DisplayHelper` when closing a screen hosting the `SceneView` composable.

Filament ≤ 1.71.5 has a race condition in `DisplayHelper`: a display-changed event (e.g. an adaptive refresh-rate switch) that is already queued when `DisplayHelper.detach()` nulls `mRenderer` is still delivered afterwards, and `updateDisplayInfo()` dereferences the nulled `mRenderer`. This was reported as google/filament#9352 and fixed by google/filament@180e91f5 ("Fix race conditions in DisplayHelper.java (#10025)"), first shipped in **Filament v1.71.6** (2026-06-10). SceneView pins 1.71.5, so every SceneView 4.x release ships the unfixed code.

SceneView 4.x is considerably more exposed to this race than 2.x. After we migrated 2.3.0 → 4.17.0 the crash rate increased ~30x/day. The reason is a teardown ordering change: in 2.3.0, `onDetachedFromSurface` ran `displayHelper.detach()` **before** `destroySwapChain` + `engine.flushAndWait()`, while the 4.x `SceneRenderer.onDetachedFromSurface` runs `flushAndWait()` **before** `detach()`. `flushAndWait()` blocks the main thread, so the refresh-rate-change event triggered by removing the rendering surface gets queued during the flush and is delivered right after `detach()` nulls `mRenderer`.

Suggested fix: bump `filament = "1.71.5"` → `"1.71.6"` in `gradle/libs.versions.toml`. Additionally/alternatively, reordering `SceneRenderer.onDetachedFromSurface` to call `displayHelper?.detach()` first (as 2.3.0 did) would shrink the race window even on older Filament versions.

### What did you expect?

Closing a screen hosting a `SceneView` composable should not crash the app.

### Steps to reproduce

```kotlin
// Timing-dependent race — not deterministically reproducible, but the pattern is:
// 1. Host the SceneView composable on a screen, rendering continuously
//    (on an adaptive-refresh device the display switches refresh rate for the rendering surface).
// 2. Close the screen (back navigation / Activity finish) so the surface is destroyed.
//    Destroying the surface makes the display switch its refresh rate back, which posts a
//    display-changed event onto the main-thread queue while
//    SceneRenderer.onDetachedFromSurface is still blocked in engine.flushAndWait().
// 3. displayHelper.detach() then nulls mRenderer, and the queued event is delivered → NPE.
// Repeatedly opening/closing the screen on an Android 16 adaptive-refresh device
// reproduces it occasionally; it shows up at scale in production crash reporting.
```

### Logs / stack trace

```text
java.lang.NullPointerException: Attempt to invoke virtual method 'com.google.android.filament.Renderer$DisplayInfo com.google.android.filament.Renderer.getDisplayInfo()' on a null object reference
    at com.google.android.filament.android.DisplayHelper.updateDisplayInfo(DisplayHelper.java:136)
    at com.google.android.filament.android.DisplayHelper$1.onDisplayChanged(DisplayHelper.java:102)
    at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.handleDisplayEventInner(DisplayManagerGlobal.java:2279)
    at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.lambda$sendDisplayEvent$0(DisplayManagerGlobal.java:2181)
    at android.os.Handler.handleCallback(Handler.java:1070)
    at android.os.Handler.dispatchMessage(Handler.java:125)
    at android.os.Looper.loopOnce(Looper.java:288)
    at android.os.Looper.loop(Looper.java:392)
    at android.app.ActivityThread.main(ActivityThread.java:10346)
```
